src/tracer.js

```diff
@@ -231,13 +231,23 @@ export default class Tracer {
             ctx.traceId = parent.traceId;
             ctx.spanId = Utils.getRandom64();
             ctx.parentId = parent.spanId;
-            ctx.flags = parent.flags;
-
             // reuse parent's baggage as we'll never change it
             ctx.baggage = parent.baggage;
+            ctx.flags = parent.flags;
 
-            parent.finalizeSampling();
-            ctx.finalizeSampling();
+            // Parent exists without a sampling decision
+            // An example of where this happens is in mobile clients where sampler parameters
+            // are not propagated
+            if (parent.flags & constants.DEFERRED_SAMPLING_MASK) {
```

Comment:
This still doesn't solve what happens if the parent context creates more than 1 child span. The tracer might be using a ratelimiting sampler which means that sampler.isSampled might return different results for 2 spans in the same trace which is not what we want.